src/utils.py

```diff
@@ -43,23 +43,46 @@ def get_events():
     return events
 
 
-def sync_organization():
-    pass
+# TODO
+# sync_members()
+
+
+def sync_customer(project_id):
+    project_data = mp.get_project(project_id=project_id)
+    for customer in wc.list_customers():
+        if project_data.attributes.organization == customer['name']:
+            return customer  # data of existing customer
+    else:
+        return wc.create_customer(name=project_data.attributes.organization,  # data of a new customer
+                                  email=project_data.owner.email,
+                                  address="Narva mnt 18",
+                                  registration_code=123,
+                                  backend_id=project_data.attributes.organization,
+                                  contact_details=project_data.attributes.organization,
+                                  country=pycountry.countries.get(name=project_data.attributes.country).alpha_2,
+                                  domain=project_data.attributes.department_webpage,
+                                  homepage=project_data.attributes.department_webpage,
+                                  native_name=project_data.owner.name,
+                                  )
 
 
 def sync_projects():
     for event in get_events():
         if event.resource == 'project':
+            project_data = mp.get_project(event.project_id)
+            customer_data = sync_customer(project_id=event.project_id)
             if event.type == 'create':
-                project_data = mp.get_project(event.project_id)
-                wc.create_project(customer_uuid="1f8643e30e424c8cbfbb960301c20fb0",  # hardcoded uuid
-                                  name=project_data.attributes.name, backend_id=project_data.id)
-                sync_orders(project_id_=event.project_id, project_name=project_data.attributes.name)
+                wc.create_project(customer_uuid=customer_data['uuid'],  # hardcoded uuid
+                                  name=project_data.attributes.name,
+                                  backend_id=project_data.id)
+                sync_orders(project_id_=event.project_id,
+                            project_name=project_data.attributes.name)
             if event.type == 'update':
                 # TODO
                 pass
             if event.type == 'delete':
                 # TODO
+                wc.delete_project(project=event.project_id)
```

Comment:
No, this one won't work because `event.project_id` is project an id from Marketplace, while WaldurClient expects to get project UUID from Waldur. Consider project filtering on its `backend_id`, i.e. `wc.list_projects({'backend_id': str(event.project_id)})`

Reply:
there are no such events (no deletion possibility of projects on eosc mp) atm, so I just will leave it be for time being